### PR TITLE
chores : Refactored token API calls into functions in frontend

### PR DESCRIPTION
**Pull request by hanzong111**
functionalities 
1) Server is able to verify integrity of incoming requests. Server will only respond to requests that comes with a ID_Token , If no Token is found , Server will force user to login again .

Changes :
1) Refactored token API calls into functions in login.js & menu.js
    (a) verify_token() => Checks if user has a valid token or not (calls verify_token API)
    (b) get_ID_Token() => get ID_Token cookie from server (calls create_token API)
2) Added verify_token before every request client makes to the server
     - our web app will force users to logout once their ID_Token expires (For testing purpose , i've set the token to have a max_age of 5s)

### CODE REVIEW

File: frontend/src/js/login.js [hunk 1]
```diff
@@ -68,6 +56,24 @@ function showLoginPage() {
     });
 }
 
+async function get_ID_Token(username, password) {
+	try 
+	{
```
Comment (dastrukar) [inline frontend/src/js/login.js]: inconsistent brace style, please fix

see how on line 59 it's
```js
async function get_ID_Token(username, password) {
```
the brace is on the same line, but for line 60-61 it's
```js
	try 
	{
```

the same applies for line 69 (nice)

File: frontend/src/js/menu.js [hunk 2]
```diff
@@ -17,18 +17,51 @@ function showMenu() {
 	`;
   }
   
-  function showUserProfile() {
-	alert('User Profile feature coming soon!');
+  async function showUserProfile() {
+	if((await verify_token()))
```
Comment (dastrukar) [inline frontend/src/js/menu.js]: i know this is a temp function, but there's an extra set of parenthesis here lol

File: frontend/src/js/menu.js [hunk 3]
```diff
@@ -17,18 +17,51 @@ function showMenu() {
...
+	{
+		alert('Please Login Again !');
+		showLoginPage();
+	}
   }
   
-  function showAddFriends() {
-	alert('Add Friends feature coming soon!');
+  async function showAddFriends() {
+	if( await verify_token())
```
Comment (dastrukar) [inline frontend/src/js/menu.js]: extra space in front of `await`?

File: frontend/src/js/login.js [hunk 4]
```diff
@@ -68,6 +56,24 @@ function showLoginPage() {
...
+		const status = await fetch('/token_management/create_token/', {
+		method: 'POST',
```
Comment (dastrukar) [inline frontend/src/js/login.js]: please indent the json data for better clarity

**Review by dastrukar — CHANGES_REQUESTED** [review]
code style is a bit inconsistent, please fix before merge, thanks

**Comment by hanzong111** [comment]
Doneeeee

**Review by dastrukar — APPROVED** [review]
looks good :]

**Pull request merged**